### PR TITLE
VIH-7376 Translation Framework

### PR DESCRIPTION
**Pull request by levibutler**
### JIRA link (if applicable) ###
VIH-7376

### Change description ###
Translation Framework
To switch to Test Language, from the console
`$('#translateToTl').click()`

To switch between en and cy from the console
`$('#translateSwitch').click()`

Feature toggled by the missing text for the translation button links.

Sample of the test language translation below
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/app.module.ts [hunk 1]
```diff
@@ -23,6 +23,16 @@ import { GlobalErrorHandler } from './shared/providers/global-error-handler';
 import { SharedModule } from './shared/shared.module';
 import { WaitingSpaceModule } from './waiting-space/waiting-space.module';
 import { ConfigSettingsResolveService } from 'src/app/services/config-settings-resolve.service';
+import { TranslateModule, TranslateLoader, MissingTranslationHandler } from '@ngx-translate/core';
+import { TranslateHttpLoader } from '@ngx-translate/http-loader';
+import { DisplayMissingTranslationHandler } from './shared/display-missing-translation-handler';
+
+export function createTranslateLoader() {
+    // We cant inject a loader because it has a race condition with adal
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/app.module.ts:31]: httpClient not loader

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-hearings/hearing-list-table.component.html [hunk 2]
```diff
@@ -28,18 +28,18 @@
...
+          *ngIf="!canStartHearing(conference)"
           (click)="signIntoConference(conference)"
-          aria-label="Sign into hearing"
+          attr.aria-label="{{ 'hearing-list-table.button.sign-into-hearing' | translate }}"
         >
-          Sign into hearing
+        {{ 'hearing-list-table.button.sign-into-hearing' | translate }}
         </button>
-        <span *ngIf="!canStartHearing(conference)">
+        <span *ngIf="canStartHearing(conference)">
```
Comment (surfsquid) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-hearings/hearing-list-table.component.html:37]: is this supposed to have been inverted? (removed leading '!')

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-hearings/hearing-list-table.component.html:37]: good spot! :) nope was for testing

**Review by surfsquid — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/test-language.service.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        console.info(`${this.loggerPrefix} Subscribing to onLanguageChange`);
+        this.translateService.onLangChange.pipe(filter(e => e.lang === 'tl')).subscribe(() => this.setupTestLanguageTranslations());
+    }
+
+    setupTestLanguageTranslations() {
+        if (!this.translationLanguageLoaded) {
+            this.translationLanguageLoaded = true;
+            this.translateService
+                .getTranslation('en')
+                .toPromise()
```
Comment (phillipjchambers) [inline VideoWeb/VideoWeb/ClientApp/src/app/shared/test-language.service.ts:22]: doesn't this work with observables?

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/test-language.service.ts:22]: covered with tests so we can try :)

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/test-language.service.ts:22]: the previous switch didnt with with tests, filter is working 🤷

**Review by phillipjchambers — APPROVED** [review]
LGTM

**Pull request merged**